events/api/src/main/java/org/projectnessie/events/api/Content.java

```diff
@@ -15,28 +15,37 @@
  */
 package org.projectnessie.events.api;
 
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import java.util.Collections;
 import java.util.Map;
 import org.immutables.value.Value;
-import org.projectnessie.events.api.json.ContentTypeIdResolver;
 
-/**
- * An object stored in Nessie, such as a table or a view.
- *
- * @see Namespace
- * @see IcebergTable
- * @see IcebergView
- * @see DeltaLakeTable
- * @see UDF
- * @see GenericContent
- */
-@JsonTypeIdResolver(ContentTypeIdResolver.class)
-@JsonTypeInfo(use = JsonTypeInfo.Id.CUSTOM, visible = true, property = "type")
+/** An object stored in Nessie, such as a table or a view. */
+@Value.Immutable
+@JsonSerialize(as = ImmutableContent.class)
+@JsonDeserialize(as = ImmutableContent.class)
 public interface Content {
 
-  ContentType getType();
+  /**
+   * Returns the type for this object.
+   *
+   * <p>Values returned here match the JSON type name used for serializing the original content
+   * object.
+   *
+   * <p>The currently-known built-in content types are:
+   *
+   * <ul>
+   *   <li>{@code ICEBERG_TABLE};
+   *   <li>{@code ICEBERG_VIEW};
+   *   <li>{@code NAMESPACE};
+   *   <li>{@code UDF};
+   *   <li>{@code DELTA_LAKE_TABLE}.
+   * </ul>
+   *
+   * Other content types may be added in the future, or registered by end users.
+   */
+  String getType();
```

Comment:
WDYT about adding something like `asModelObject()` for auto-conversion to model API java wrappers?

Clients that are sure to have the right custom types deployed, could benefit by not having to maintain the code do this conversion on their side.

Reply:
That would require a dependency on the `nessie-model` module, not sure if that is a good thing. We could maybe provide an utility class somewhere else. @snazy wdyt?

OTOH, I'm not sure how much support we want to provide for back-and-forth conversion between Event and Model. Smart users will certainly use their own schemas when processing events, because relying on Nessie's Event schema means that they are tying their business logic to Nessie's domain. So in my opinion if users need to retrieve the original Model object, they will probably have to do so by deserializing some user-specific domain object that was created from the original Event object from Nessie.